### PR TITLE
Update Smithy types to 3.0.0 and AWS SDK to 3.577.0

### PR DESCRIPTION
**Pull request by douddle**
Last version of AWS SDK is using Smithy types 3.0.0, no more 1.1.0.
So let's try this bump
Is it related to this issue : https://github.com/m-radzikowski/aws-sdk-client-mock/issues/197 ?

### CODE REVIEW

**Comment by douddle** [comment]
I don't know why but Actions does not pull my amended commit with fixed comment.

**Comment by lynxSven** [comment]
For the PR lint task change the name of the PR to --> chore: Update Smithy types to 3.0.0 and AWS SDK to 3.577.0

**Comment by douddle** [comment]
> For the PR lint task change the name of the PR to --> chore: Update Smithy types to 3.0.0 and AWS SDK to 3.577.0

Updated but still failing. I do't know why `chore: ` is dropped during check

**Comment by m-radzikowski** [comment]
Hey,
thank you for the PR. However, those are only dev dependencies, used in unit tests. The `aws-sdk-client-mock` works fine with both `@smithy/types` v1.1.0 and 3.0.0.

When you install the `aws-sdk-client-mock`, it does not install `@smithy/types`. They are installed as dependency of `@aws-sdk/*` packages. The `aws-sdk-client-mock` deliberately does not list it as a dependency to reduce the chance of installing a conflicting version, but it still happens because `@aws-sdk/*` packages in different versions can install different types - hence the steps described in #197

The AWS SDK version 3.363.0 used here as dev dependency is the one that introduced `@smithy/types`. Since `aws-sdk-client-mock` v3+ is compatible with AWS SDK v3.363.0+, I want to run unit tests on v3.363.0 to prevent accidental breaking changes.

So, in short - thanks, but this does not affect lib usage or solve #197 and I don't want to bump it.
